src/bin/tikv-ctl.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(plugin)]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+extern crate tikv;
+extern crate clap;
+extern crate protobuf;
+extern crate kvproto;
+extern crate rocksdb;
+
+use std::{str, u64};
+use clap::{Arg, App, SubCommand};
+use protobuf::Message;
+use kvproto::raft_cmdpb::RaftCmdRequest;
+use kvproto::raft_serverpb::{RaftLocalState, RegionLocalState, RaftApplyState};
+use kvproto::eraftpb::Entry;
+use rocksdb::DB;
+use tikv::util::{self, escape, unescape};
+use tikv::raftstore::store::keys;
+use tikv::raftstore::store::engine::{Peekable, Iterable};
+use tikv::storage::{ALL_CFS, CF_RAFT};
+
+static VERSION: &'static str = "0.0.1";
+const DEFAULT_TO: Vec<i32> = vec![0xff];
+
+fn main() {
+    let matches = App::new("tikv")
+        .version(VERSION)
+        .author("PingCAP")
+        .about("Distributed transactional key value database powered by Rust and Raft")
+        .arg(Arg::with_name("db")
+             .short("d")
+             .takes_value(true)
+             .help("set rocksdb path, required"))
+        .subcommand(SubCommand::with_name("raft")
+                    .about("print raft log entry")
+                    .version(VERSION)
+                    .subcommand(Subcommand::with_name("log")
+                                .about("print the raft log entry info")
+                                .version(VERSION)
+                                .arg(Arg::with_name("region")
+                                     .short("r")
+                                     .takes_value(true)
+                                     .help("set the region id, required when get raft message"))
+                                .arg(Arg::with_name("index")
+                                     .short("i")
+                                     .takes_value(true)
+                                     .help("set the raft log index")))
+                    .subcommand(SubCommand::with_name("range")
```

Comment:
what I mean is:

+ ` .subcommand(SubCommand::with_name("range")` is not only for `raft`, `range` can be used in other places
+ `.subcommand(SubCommand::with_name("region")` should be in `raft`, `region` is only used in `raft`